### PR TITLE
comments out crafting frag-12 shells

### PR DESCRIPTION
**Pull request by AdamElTablawy**
For those not in the know, frag-12 shells are gyrojet rounds for your shotgun.

When they hit you, they explode, knocking you out, and possibly delimbing you. They are as a result superior to every other shotgun shell you can make and also every other gun in the game, and they're easy enough to create, as they only need tech shells and easily available chemicals. The hardest one to get is from corn, and you can get corn seeds quite easily.

To anyone with even half a mind, they have infinite instant-KO hardstun shells that can quite possibly crit you on the first hit. Aside from shielded hardsuits, nothing will block this.

They shouldn't be _craftable_. If you want to make them cargo orderable, or something, be my guest. 
## Changelog
:cl:
del: Removed craftable frag-12 rounds. 
/:cl:

### CODE REVIEW

**Comment by datlo** [comment]
Neutral on this because I do agree that they are OP. I'd suggest removing the code for crafting entirely, we dont need more commented out code in the codebase. If anyone wants to revert this in the future thats what we have git for.

**Comment by MRgaty** [comment]
I rarely see them being used as it requires departmental cooperation to have it made and a decent amount of work just to have some ammo for sec.
Also making them avaliable in cargo would be even worse, I dont like this.

**Comment by Mitchs98** [comment]
Disagree with this. It's not something you can just magic up on a whim. It takes solid co-operation and a fair time investment to even get them. Not to mention getting enough to make them viable for extended combat. I rarely seem them used as Gaty said, much less abused to any extent.

**Comment by AdamElTablawy** [comment]
Since when has a minor time investment and growing _corn_ been a justification for instant kills?

Likewise, that it hasn't been abused _yet_ is not a justification for keeping it so that when it IS abused nobody has any idea what to do.

You said rarely. Therefore, it has been abused. Therefore, it should be removed before it becomes more commonplace.

They ARE something you can magic up on a whim, and involves no cooperation or time investment.

It does not take 'interdepartmental cooperation'. It takes going to the minigarden or asking the botanist for one corn seed, then growing the corn yourself, in science, to shit out as many frag 12 shells as you want.

It's incredibly simple to obtain. Techshells are obtained nearly immediately, and all its ingredients are roundstart. That it is not abused is purely because people don't know how powerful it is.

There is no argument in favor of keeping it that I have seen here, just variations of 'well it takes effort though.'

Which, they don't. Mechs take effort. Adv eguns take effort. Xray implants take effort. These take absolutely nothing but -corn- and minor patience. Mineral requirement nil. 

And these are _not_ for security, they're for any enterprising scientist who makes an improvised shotgun, which they can craft immediately, by the way, and then claims the shells are for 'testing.' The typical excuse, and one that works, consistently. Or if the threat is bad enough, like warops, no excuse required-just whip out the instant kill cannon and call it a day.

They're stronger than most weapons in a traitor's arsenal, or anything else in science's arsenal, bar mechs. (And even then, a few of these  will drop a gygax.) There's nothing that can resist or counter them aside from total deflect things like shielded hardsuits, and even those won't protect you from the blast.

They're far, far too powerful to be items the crew can mass produce, and given they take five units of glycerol, an easy to obtain chem gotten from grinding corn, which you can grow yourself in science in a hydroponics tray you build and upgrade with minimal effort, there is no teamwork, no collaboration, and no balance.

If we want to pretend that asking the botanist to give you a corn seed is 'solid co-operation' and justifies science having access to gyrojet rounds, then I can just give botany gatfruit seeds that must be planted by a security officer to grow and call it a day. Otherwise, these are self-evidently overpowered as shit and no amount of endlessly repeating 'interdepartmental cooperation' will change that.

The alternative would be to remove the explosion they cause entirely, or make them ungodly hard to make, because the only other thing of comparable power is the SRM missile rack for mechs and THAT is a solid time investment.

**Comment by Fox-McCloud** [comment]
These almost never come up and require cross department co-operation---I question how urgent it is for these to be removed.

**Comment by Mitchs98** [comment]
> Likewise, that it hasn't been abused _yet_ is not a justification for keeping it so that when it IS abused nobody has any idea what to do.

If you want to go about removing things in this manner, you may as well remove half the game, including items that are MUCH EASIER to obtain that are or may as well be instakills and aren't traitor only.

> You said rarely. Therefore, it has been abused. Therefore, it should be removed before it becomes more commonplace.

> I rarely seem them used as Gaty said, much less abused to any extent.

Actually, I said I rarely see them _used_, and have never seen them _abused_. Please don't entirely misquote what I say.

> They ARE something you can magic up on a whim, and involves no cooperation or time investment.

It involves a lot of time investment if you want any feasible amount of the shells to use, shells that can easily miss mind you, to use for any extended amount of time. Growing the corn does take time, a decent amount of time, especially if you want 100 potency corn that makes mass producing these viable. Saying they require no time investment whatsoever is, quite frankly, entirely untrue. Default corn only has 0.2units of corn oil when ground up, it takes 3 FULL units of corn oil plus sacid to make a single unit of Glycerol. 

> It does not take 'interdepartmental cooperation'. It takes going to the minigarden or asking the botanist for one corn seed, then growing the corn yourself, in science, to shit out as many frag 12 shells as you want.
> 
> It's incredibly simple to obtain. Techshells are obtained nearly immediately, and all its ingredients are roundstart. That it is not abused is purely because people don't know how powerful it is.

Except it's not abused because it's not incredibly simple to maintain. In theory it is, in practice it takes a long time when there are easier options available. It's not abused because it's incredibly time consuming.

> There is no argument in favor of keeping it that I have seen here, just variations of 'well it takes effort though.'
> 
> Which, they don't. Mechs take effort. Adv eguns take effort. Xray implants take effort. These take absolutely nothing but -corn- and minor patience. Mineral requirement nil.

On the list you mention, mechs are the only thing that take even a modicum of effort comparable to manufacturing frag-12 shells, and that's only because it actually has a building process. AEG's and X-Rays only take 'effort' in the form of cramming items in a machine and bitching at mining for materials. Not quite sure how that's more effort than tending to a garden, growing a high potency corn, and mixing Glycerol on top of crafting the shells through the crafting menu one at a time.

> And these are _not_ for security, they're for any enterprising scientist who makes an improvised shotgun, which they can craft immediately, by the way, and then claims the shells are for 'testing.' The typical excuse, and one that works, consistently. Or if the threat is bad enough, like warops, no excuse required-just whip out the instant kill cannon and call it a day.

That's when admin intervention comes into play, at-least for the former example. For the latter example Warops is mostly entirely fair game with everyone being allowed to do nearly anything(within relative reason and rule reasoning) to stop the nukies. And despite this I have never, in the entire 2k+ hours I've played here, witnessed _anyone_ 'abuse' frag-12 against nukies.

> They're far, far too powerful to be items the crew can mass produce, and given they take five units of glycerol, an easy to obtain chem gotten from grinding corn, which you can grow yourself in science in a hydroponics tray you build and upgrade with minimal effort, there is no teamwork, no collaboration, and no balance.

`reagents_add = list("cornoil" = 0.2, "vitamin" = 0.04, "plantmatter" = 0.1)`
Actually, corn produces cornoil when ground, which must further be made into Glycerol at a 3 cornoil 1 glycerol ratio.

> If we want to pretend that asking the botanist to give you a corn seed is 'solid co-operation' and justifies science having access to gyrojet rounds, then I can just give botany gatfruit seeds that must be planted by a security officer to grow and call it a day. Otherwise, these are self-evidently overpowered as shit and no amount of endlessly repeating 'interdepartmental cooperation' will change that.

A hyperbole statement that falls flat to the rest of the argument. Gatfruit grow fully loaded .357 infinitely so long as your seeds don't die/you don't run out of seeds. THEY take no effort to produce, whilst Frag-12 requires prep time and quite a few other components. You need 1. A shotgun of some kind. 2. Science access/a scientist willing to print a large amount of tech shells and materials for Frag-12 for you. 3. High potency corn to make it not take five years to grow enough corn to make a reasonable amount of shells. 4. A chemistry machine to produce sulfuric acid.

All of which adds up to far more effort than a mech or anything else on your list that you counter claim as to taking more effort to produce.

**Comment by AdamElTablawy** [comment]
@Fox-McCloud 

Breaking into botany and grabbing corn seeds is not co-operation.

They take no effort.

And just because it doesn't come up doesn't mean it's not imbalanced and WILL propose a balance issue eventually.

How long did it take before hellhounds finally became a balance issue and we had to remove them?

I'm not even going to bother reading whatever the fuck mitch wrote tbh

**Comment by Quantum-M** [comment]
You do have to take in consideration that, not taking in account of the corn seeds and some materials cost, science is capable of getting most getting almost everything up to get such chemical, and that science already has a lot of access, especially as a traitor with a emag, to a lot of weapons, considering that:

- RnD is usually done by the 30 mark, with the botany equipment usually available by 20 minutes in.
- Science has access to their chemistry, which can make the rest of the glycerol recipe, along with having mutagen for any corn they grow.
- You need at the very least trays and the seed making machine for mass production, and if you have more than one tray, and one harvest leads to more corn being made, you would use the seeds of that harvest and so forth.
- The Arrival Gardens has a chance to have corn seeds them.
- Botany role, along with several roles, tend to be very lenient to giving stuff out, especially  when it comes to something as harmless sounding as corn, and if they don't hand it out, science has and will use tools to get said seeds.

To add to this, should _non-antags_ have the means to have access to weapons that are instant kill/near instant kill levels? Considering that in cases that it would be "legitimate" to use, it would be during Nuke Ops or Wizards, which by then there isn't enough time to get everything ready, or the most of the people capable of creating said stuff are hunting down said antags.

**Comment by Ionward** [comment]
Please realize asking people to upvote your PR in the discord does not look good nor does it do anything having upvotes.


Anyways. This just sounds like a "I DED PLS NRF" sort of PR. In my over four years of playing on Paradise I have **never** seen anyone use the Frag-12 shells. People already never use tech-shell ammo because of how obnoxious it is to craft, and these ones are extremely annoying. SS13 is not a game about everything being fair, either. Take a look at anything the wizard does, for example. And rules still apply to people making these shells, its not like just because they're craftable they give you the self-antag pass.

**Comment by Ionward** [comment]
> 
> 
> I'm not even going to bother reading whatever the fuck mitch wrote tbh, it's probably something along the lines of not overpowered because you have to talk to botany and you have to grow the corn or something like that takes more than ten minutes, or something like you need a shotgun to put it in which is pretending impro shotguns don't exist.


And you wanna be confrontational about people reviewing your PR, too. Come on, man.

**Comment by Mitchs98** [comment]
I didn't pretend anything didn't exist, for the record. And if your argument for removing it is the fact that it's extremely abusable, then yes you'd have to take a lot of the games content under examination for removal or adjustment. It's not hyperbole when it's entirely true.

**Comment by AdamElTablawy** [comment]
All of these just still aren't valid arguments. "I've never seen anyone abuse this broken mechanic" is not a justification that it remain and stay broken.

WHY should it remain? What benefit does it bring, that you can print instant kill bullets?

**Comment by Mitchs98** [comment]
I had a better justifying post, but you refused to read it because it countered part of your arguments/you just didn't feel like reading something that might be right. If you're not going to read posts by people why should they try to make justifiable counterpoints to your PR?

**Comment by AdamElTablawy** [comment]
I didn't read it because I never read mitchposts.

**Comment by Fox-McCloud** [comment]
@AdamElTablawy cut the attitude statements like this have no place on the git:

>I'm not even going to bother reading whatever the fuck mitch wrote tbh

**Comment by MRgaty** [comment]
*I said rarely as to "I have never seen them being used by anone in my entire playtime, I have only heard about them once when henk and mitch told me they existed"
Also using something DOES NOT have to mean they have been "abused", using something is just... Well using them.
*botany is known to be a newbie friendly job, so it means that a lot of times the job is either not working at all or working at the bare basics, you could ask for the seeds yourself but honestly making s tray (or stealing one, up to you) and wait for the plant to grow, feeding it and watering it, taking care of the stats, then harvest it and process it, then print the ammo, then use the crafting menu tho craft a round that, unless you somehow have a shotgun (And are antag, because running around as scientist with a shotgun would be iffy as heck unless nukies, in which case I dont think you will have enough time to do all this process) they will go to security where they MAY or MAY not be used.
You get the idea? You invested a quite decent amount of time and effort to print something that is not gonna give you any advantage in particular, thats why almost. 
anyone uses them and why in my opinion they should be good.
*About "instakills" that is just an exaggeration, they may knock you out yes but any taser does stun you too.
*Them being better than most nukie weapons is also an exaggeration, as you literally have grenade launchers, rifles and even mechs that will laugh at that meteor shot, which is fine.
*crafting AEGs is pressing a button if you have minerals, which is the whole purpose of mining, botany is not obligated in any way to give you anything.


In summary, just in case you dont want to read a long ass comment, I think some of your statements are just exaggerations and I have never seen them or heard anyone complaining about them. They also require a good amount of time and work to get so I think they are fine.

**Comment by Fox-McCloud** [comment]
also, unironically thumbing up your own PR:


img

**Comment by procdrone** [comment]
I personally would welcome to see some sort of rebalance _(cost wise, access wise, or stats wise)_ to tech shells, they have access to some very powerful ammunition, like pulse (CC and DS level!!!), **gyrojet**  _(first time i hear about it too)_, stun shells, etc..

True, i almost never see tech shells being used, but when i do, it always ends with the opposing person being instantly shut down, and shotguns with normal ammunition are already quite powerful.

**Comment by AdamElTablawy** [comment]
@MRgaty

No, they instakill. They’re not meteorshot. They’re gyrojet rounds.

They cause an explosion on the target. It’s akin to c4 going off on top of you. You either die instantly or are incapacitated and in crit.

They’re on par or better than most nukie weapons. The nukie grenade launcher, while extremely powerful, has much more expensive ammunition. The c20 and so forth are peashooters in comparison because they don’t instantly crit you on hit. The only concession worth making is that it’s not as powerful as shieldedsuit-desword-magboots, and even then, the blast will penetrate a shielded suit if you fire next to them instead of at them. It’s also not quite as useful as the l6 saw.

It’s certainly more powerful than everything else. No exaggerations there-one shot from this immediately floors you and puts you in hardcrit.

Botany irrelevant. Just take corn from the seed garden in arrivals and grow it yourself. If you’re an enterprising powergamer or an antag it’s the simplest task in the world.

Everything else you said is just gibberish. Scientists can make impro shotguns. Sec might use the frag rounds or might not, but if they do, there’s no counter other than just not getting hit. You invested about half an hour into massproducing some very powerful slugs that will immediately game end whoever they hit IF they’re used, so saying ‘well it’s only if they’re used’ is a lot like saying ‘well it’s only a 30% chance to die instantly.’

If someone is smart enough to use these, there’s no counter. I’ve made no exaggerations, Gaty. It’s not my fault you don’t bother to look at the code or the crafting menu so you’ve never heard of them, and it’s certainly not my fault you didn’t bother to read what they actually do and think they’re meteorshot.

They’re not meteorshot. They’re bombs. Gyrojet rounds. Science can print gyrojets.

Does that sound good to you?

**Comment by farie82** [comment]
Reading on what the projectile actually does:
25 flat brute damage on hit (armor can reduce it)
5 weaken on hit.
-1 range devastation, 0 range heavy impact and then 1 range light impact
Currently 0 range in explosive code means that it won't do anything. It's a `<` check in stead of a `<=` check for the distance.
So it's a light impact: `3` severity.

Going to `human/ex_act()`
you'll get 30 additional bruteloss unless your bomb armor check blocks it. Then only 15 extra
You will get a 50% chance to drop 1 limb.
You will get 15 ear damage and go deaf for 60 ticks.
You have a 50% chance to fall asleep for 10 ticks.

The falling asleep part is the strongest part of the bullet I'd say here.

Either way I myself don't care either way if it's removed or not but do keep to our template on why a PR should go through and what not. Use facts and reasons in the PR description. And most important stay friendly to people responding to your PR. We're all here enjoying our little space game so do be kind.

**Comment by variableundefined** [comment]
I also object to FRAG-12's removal. Closing due to two maintainer objection.

**Comment by AdamElTablawy** [comment]
Good to see paradise is about as amicable to basic balance as ever. Don’t come crying to me when some powergamer starts abusing these a year later.

**Comment by variableundefined** [comment]
Chill Chief Designer

**Comment by AffectedArc07** [comment]
Just dodge the shots lmao

**Comment by AffectedArc07** [comment]
> Good to see paradise is about as amicable to basic balance as ever. Don’t come crying to me when some powergamer starts abusing these a year later.

Pleased to report almost 2 years later, these are still never used.

Verdict: 🧂
